src/WebJobs.Script.WebHost/Diagnostics/DeferredLogger/DeferredLogMessage.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
+{
+    public class DeferredLogMessage
+    {
+        public DateTimeOffset Timestamp { get; internal set; }
```

Comment:
I want the logic for figuring out the function name to come from the Loggers themselves. This is just trying to get all of those details down to the other logger so it can do all of its figuring.